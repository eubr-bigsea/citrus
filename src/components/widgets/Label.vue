<template>
    <span class="label">
        <span :data-order="field.order" :data-name="field.name"
            :class="{'text-danger': error, required: field.required}" :title="error">{{field.label || field.name}}</span>
        <span v-show="field.required">*</span>:
        <span v-if="(showHelp === undefined || showHelp) && field.help" class="data-help">
            <span class="fa fa-question-circle float-right" :title="field.help"></span>
        </span>
    </span>
</template>

<script>
    import Widget from '../../mixins/Widget.js';
    export default {
        mixins: [Widget],
        name: "property-label",
        props: {
            field: { required: true, type: Object },
            value: {},
            showHelp: true,
        },
        computed: {
            error() {
                if (this.field.required && (this.value === undefined || this.value === null || this.value === '' || this.value === {})) {
                    return this.$tc('errors.missingRequiredValue');
                }
                return null;
            },
        }
    }
</script>
<style scoped>
    .label {
        font-size: 1em;
    }
</style>
