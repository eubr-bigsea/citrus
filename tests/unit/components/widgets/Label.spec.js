import { shallowMount, createLocalVue } from '@vue/test-utils';
import Label from '$SRC/components/widgets/Label.vue';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import Vue from 'vue';
Vue.component('FontAwesomeIcon', FontAwesomeIcon);

describe('Label.vue', () => {
    const localVue = createLocalVue();
    localVue.component("FontAwesomeIcon", FontAwesomeIcon);


    it('renders a required field label', () => {
        const text = 'Please, inform required item';
        const wrapper = shallowMount(Label, {
            localVue,
            propsData: {
                field: {
                    label: text,
                    required: true
                }
            },
            mocks: {
                $t: (msg) => msg,
                $tc: (msg) => msg,
            }
        });
        const label = wrapper.find('[class="label"]');
        expect(label.exists()).toBeTruthy();
        // xa0 = &nbsp;
        expect(label.text()).toBe(`${text}\xa0*:`);
        const labelHelp = wrapper.find('[class="data-help"]');
        expect(labelHelp.exists()).toBeFalsy();

    });
    it('renders label with help', () => {
        const text = 'Inform your name';
        const help = 'Inform your name, as recorded';
        const wrapper = shallowMount(Label, {
            propsData: {
                localVue,
                field: {
                    label: text,
                    required: false,
                    help
                },
                showHelp: true
            }
        });
        const label = wrapper.find('[class="label"]');
        expect(label.exists()).toBeTruthy();
        expect(label.text()).toBe(`${text}:`);
        const labelHelp = wrapper.find('[icon*="fa"]');
        expect(labelHelp.exists()).toBeTruthy();
        expect(labelHelp.attributes('title')).toBe(help);
    });  
});
