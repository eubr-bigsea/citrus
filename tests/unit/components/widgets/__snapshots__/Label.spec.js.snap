// Vitest Snapshot v1

exports[`Integer.vue > snapshots the component 1`] = `""`;

exports[`Label.vue > snapshots the component 1`] = `":"`;
